### ISSUE TITLE
SyntaxError: Invalid regular expression

### ISSUE THREAD

**Issue by lemmingapex** [tristen/suggestions#20]
Line 192 can throw syntaxErrors when query results contains regex special characters:
https://github.com/tristen/suggestions/blob/gh-pages/src/suggestions.js#L192

for example a query result of: "test ( test"

throws:
suggestions.js:192 Uncaught SyntaxError: Invalid regular expression: /(test( test)/: Unterminated group

In order to fix this, you might consider not using a regular expression to bold the query text.  A faster and simpler solution would be to use indexOf and slice.  Something like this:

```
this.data.map(function(d) {
  var boldString = this.getItemValue(d);
  var indexOfQuery = boldString.indexOf(this.query);
  if(indexOfQuery > -1) {
    var endIndexOfQuery = indexOfQuery + this.query.length();
    boldString = boldString.slice(0, endIndexOfQuery) + '</strong>' + boldString.slice(endIndexOfQuery)
    boldString = boldString.slice(0, indexOfQuery) + '<strong>' + boldString.slice(indexOfQuery)
  }
  return {
    original: d,
    string: boldString
  };
}.bind(this));
```

I have not tested or run the above code.

If you do fix these issues, please add appropriate tests.

For future reference, if you need to escape special characters in a string, take a look at:
https://lodash.com/docs/4.17.4#escapeRegExp

**Comment by lemmingapex** [comment]
@tristen What do you think of the above?  When are looking to release v1.3.3?

**Comment by tristen** [comment]
@lemmingapex I'd happily accept a PR for this change. The solution sounds 👍  but I'm not sure when I'll have time to do it.

**Comment by tristen** [comment]
Closed via #21
